compiler/packages/babel-plugin-react-compiler/src/Entrypoint/Program.ts

```diff
@@ -417,39 +417,39 @@ export function compileProgram(
         compileResult = {kind: 'error', error: err};
       }
     }
-    // If non-memoization features are enabled, retry regardless of error kind
-    if (compileResult.kind === 'error' && environment.enableFire) {
+
+    if (compileResult.kind === 'error') {
+      /**
+       * If an opt out directive is present, log only instead of throwing and don't mark as
+       * containing a critical error.
+       */
+      if (optOutDirectives.length > 0) {
+        logError(compileResult.error, pass, fn.node.loc ?? null);
+      } else {
+        handleError(compileResult.error, pass, fn.node.loc ?? null);
+      }
```

Comment:
Moved this up as we want to report / log errors *before* retrying for (1) `panicThreshold: 'all' | 'critical_error'` modes and (2) to avoid overwriting bailout logging for non-Fire functions.